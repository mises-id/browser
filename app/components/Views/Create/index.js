/*
 * @Author: lmk
 * @Date: 2021-07-19 12:17:48
 * @LastEditTime: 2021-07-19 18:10:32
 * @LastEditors: lmk
 * @Description: Restore misesid
 */
import { strings } from 'app/locales/i18n';
import React, { useReducer, useEffect, useRef, useState } from 'react';
import {Platform, StyleSheet, Text, View,Image ,TouchableOpacity, TextInput, Button, ScrollView} from 'react-native';
const word = (item={},key)=>{
  return <View key={key} style={styles.wordContent}>
    <Text style={styles.wordTxt}>{item.value}</Text>
    <View style={styles.isHidden}>
      <Text style={styles.isHiddenTxt}>?</Text>
    </View>
  </View>
}
const Create = (props)=>{
  const [data, setdata] = useState([{value:'cloud'},{value:'rabbit'},{value:'fly'},{value:'hawk'},{value:'contemplate'},{value:'pretty'},{value:'superduty'},{value:'lovely'},{value:'come'},{value:'gift'},{value:'main'},{value:'green'}])
  return <ScrollView>
    <View style={styles.pageBox}>
    <View style={styles.titleBox}><Text style={styles.title}>{strings('restore.title')}</Text></View>
    <View style={styles.inputBox}>
      <TextInput placeholder={strings('common.placeholder')}></TextInput>
    </View>
    <View style={styles.wordsContainer}>
      {data.map(word)}
    </View>
    <View style={styles.btnBox}>
      <View style={[styles.btnStyle,styles.success]}>
        <Text style={styles.successBtnTxt}>{strings('create.success_button')}</Text>
      </View>
      <View style={[styles.btnStyle,styles.cancel]}>
        <Text style={styles.cancelBtnTxt}>{strings('common.cancel_button')}</Text>
      </View>
      <View style={styles.back}><Text style={styles.backTxt}>{strings('create.back')}</Text></View>
    </View>
  </View>
  </ScrollView>
}
const styles = StyleSheet.create({
  pageBox:{
    paddingLeft:15,
    paddingRight:15
  },
  title:{
    fontSize:16,
    color:"#333",
    fontWeight:"bold"
  },
  titleBox:{
    paddingTop:25,
    paddingBottom:17
  },
  inputBox:{
    height:50,
    backgroundColor:"#F8F8F8",
    borderRadius:5,
    paddingLeft:15,
    paddingRight:15,
    marginBottom:25
  },
  wordsContainer:{
    borderColor:"#EEEEEE",
    borderRadius:5,
    borderWidth:1,
    flexDirection: 'row',
    flexWrap:"wrap",
    padding:5
  },
  wordContent:{
    padding:15,
    borderRadius:5,
    position:'relative',
    alignItems:'center',
    justifyContent:'center',
    backgroundColor:'#F7F7FF',
    margin:5,
  },
  wordTxt:{
    fontSize:16,
    fontWeight:'bold',
    color:'#5D61FF'
  },
  btnBox:{
    marginTop:15,
    marginLeft:25,
    marginRight:25
  },
  btnStyle:{
    borderWidth:1,
    height:50,
    borderRadius:50,
    alignItems:'center',
    justifyContent:'center',
    overflow:'hidden',
    marginTop:25
  },
  success:{
    borderColor:'#5D61FF'
  },
  successBtnTxt:{
    fontSize:16,
    color:"#5D61FF",
    fontWeight:'bold'
  },
  cancel:{
    borderColor:'#EEEEEE',
    backgroundColor:'#EEEEEE'
  },
  cancelBtnTxt:{
    fontSize:16,
    color:"#666",
    fontWeight:'bold'
  },
  isHidden:{
    position:"absolute",
    left:0,
    right:0,
    top:0,
    bottom:0,
    backgroundColor:'#eee',
    borderRadius:5,
    alignItems:"center",
    justifyContent:"center"
  },
  isHiddenTxt:{
    color:'#999',
    fontSize:21
  },
  back:{
    marginTop:25,
    alignItems:'center',
    marginBottom:200
  },
  backTxt:{
    fontSize:15,
    color:'#5D61FF'
  }
})
Create.navigationOptions = ()=>{
  return {
    headerTitle:strings('menu.create')
  }
}
export default Create