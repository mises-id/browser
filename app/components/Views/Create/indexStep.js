/*
 * @Author: lmk
 * @Date: 2021-07-19 12:17:48
 * @LastEditTime: 2021-08-14 18:23:35
 * @LastEditors: lmk
 * @Description: Restore misesid
 */
import Sdk from 'app/core/Sdk';
import {strings} from 'app/locales/i18n';
import React, {useEffect, useState} from 'react';
import {StyleSheet, Text, View, TouchableOpacity} from 'react-native';
const word = (item = {}, key) => {
  return (
    <View key={key} style={styles.wordContent}>
      <Text style={styles.wordTxt}>{item}</Text>
    </View>
  );
};
const CreateStep = ({navigation}) => {
  const [data, setdata] = useState([]);
  const [mnemonics, setmnemonics] = useState('');
  useEffect(() => {
    Sdk.randomMnemonics().then(res => {
      const Word = res.split(' ');
      setdata(Word);
      setmnemonics(Word.join(','));
    });
  }, []);
  const nextStep = () => {
    navigation.push('CreateStep2', {
      mnemonics,
    });
  };
  return (
    <View style={styles.pageBox}>
      <View style={styles.titleBox}>
        <Text style={styles.title}>{strings('create.Step1Title')}</Text>
      </View>
      <View style={styles.desc}>
        <Text style={styles.title}>{strings('create.desc')}</Text>
      </View>
      <View style={styles.wordsContainer}>{data.map(word)}</View>
      <View style={styles.btnBox}>
        <TouchableOpacity onPress={nextStep}>
          <View style={[styles.btnStyle, styles.success]}>
            <Text style={styles.successBtnTxt}>
              {strings('create.step_btn')}
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );
};
const styles = StyleSheet.create({
  pageBox: {
    paddingLeft: 15,
    paddingRight: 15,
  },
  title: {
    fontSize: 16,
    color: '#333',
    fontWeight: 'bold',
  },
  titleBox: {
    paddingTop: 25,
    paddingBottom: 17,
  },
  desc: {
    fontSize: 15,
    color: '#333',
    marginBottom: 25,
  },
  wordsContainer: {
    borderColor: '#EEEEEE',
    borderRadius: 5,
    borderWidth: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    padding: 5,
  },
  wordContent: {
    padding: 15,
    borderRadius: 5,
    position: 'relative',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#F7F7FF',
    margin: 5,
  },
  wordTxt: {
    fontSize: 16,
    fontWeight: 'bold',
    color: '#5D61FF',
  },
  btnBox: {
    marginTop: 15,
    marginLeft: 25,
    marginRight: 25,
  },
  btnStyle: {
    borderWidth: 1,
    height: 50,
    borderRadius: 50,
    alignItems: 'center',
    justifyContent: 'center',
    overflow: 'hidden',
    marginTop: 25,
  },
  success: {
    borderColor: '#5D61FF',
  },
  successBtnTxt: {
    fontSize: 16,
    color: '#5D61FF',
    fontWeight: 'bold',
  },
  cancel: {
    borderColor: '#EEEEEE',
    backgroundColor: '#EEEEEE',
  },
  cancelBtnTxt: {
    fontSize: 16,
    color: '#666',
    fontWeight: 'bold',
  },
});
CreateStep.navigationOptions = () => {
  return {
    headerTitle: strings('menu.create'),
  };
};
export default CreateStep;
