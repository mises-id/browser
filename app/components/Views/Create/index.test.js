/*
 * @Author: lmk
 * @Date: 2021-07-19 22:12:23
 * @LastEditTime: 2021-08-14 18:34:47
 * @LastEditors: lmk
 * @Description: Create test
 */
import React from 'react';
import {shallow} from 'enzyme';
import Create from './';

describe('Create', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Create />);
    expect(wrapper).toMatchSnapshot();
  });
});
