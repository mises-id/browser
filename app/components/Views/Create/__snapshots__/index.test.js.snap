// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create should render correctly 1`] = `
<ScrollView
  keyboardShouldPersistTaps="always"
>
  <View
    style={
      Object {
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <View
      style={
        Object {
          "paddingBottom": 17,
          "paddingTop": 25,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#333",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Enter the mnemonic words in order
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderRadius": 5,
          "height": 50,
          "marginBottom": 25,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }
    >
      <TextInput
        onChange={[Function]}
        placeholder="Please enter the contents"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "borderColor": "#EEEEEE",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "padding": 5,
        }
      }
    >
      <View
        key="0"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7FF",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "padding": 15,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D61FF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          cloud
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 5,
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#999",
                "fontSize": 21,
              }
            }
          >
            ?
          </Text>
        </View>
      </View>
      <View
        key="1"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7FF",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "padding": 15,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D61FF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          rabbit
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 5,
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#999",
                "fontSize": 21,
              }
            }
          >
            ?
          </Text>
        </View>
      </View>
      <View
        key="2"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7FF",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "padding": 15,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D61FF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          fly
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 5,
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#999",
                "fontSize": 21,
              }
            }
          >
            ?
          </Text>
        </View>
      </View>
      <View
        key="3"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7FF",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "padding": 15,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D61FF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          hawk
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 5,
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#999",
                "fontSize": 21,
              }
            }
          >
            ?
          </Text>
        </View>
      </View>
      <View
        key="4"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7FF",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "padding": 15,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D61FF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          contemplate
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 5,
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#999",
                "fontSize": 21,
              }
            }
          >
            ?
          </Text>
        </View>
      </View>
      <View
        key="5"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7FF",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "padding": 15,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D61FF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          pretty
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 5,
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#999",
                "fontSize": 21,
              }
            }
          >
            ?
          </Text>
        </View>
      </View>
      <View
        key="6"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7FF",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "padding": 15,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D61FF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          superduty
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 5,
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#999",
                "fontSize": 21,
              }
            }
          >
            ?
          </Text>
        </View>
      </View>
      <View
        key="7"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7FF",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "padding": 15,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D61FF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          lovely
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 5,
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#999",
                "fontSize": 21,
              }
            }
          >
            ?
          </Text>
        </View>
      </View>
      <View
        key="8"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7FF",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "padding": 15,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D61FF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          come
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 5,
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#999",
                "fontSize": 21,
              }
            }
          >
            ?
          </Text>
        </View>
      </View>
      <View
        key="9"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7FF",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "padding": 15,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D61FF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          gift
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 5,
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#999",
                "fontSize": 21,
              }
            }
          >
            ?
          </Text>
        </View>
      </View>
      <View
        key="10"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7FF",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "padding": 15,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D61FF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          main
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 5,
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#999",
                "fontSize": 21,
              }
            }
          >
            ?
          </Text>
        </View>
      </View>
      <View
        key="11"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F7FF",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "padding": 15,
            "position": "relative",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D61FF",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          green
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 5,
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#999",
                "fontSize": 21,
              }
            }
          >
            ?
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginLeft": 25,
          "marginRight": 25,
          "marginTop": 15,
        }
      }
    >
      <TouchableOpacity
        onPress={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 50,
                "borderWidth": 1,
                "height": 50,
                "justifyContent": "center",
                "marginTop": 25,
                "overflow": "hidden",
              },
              Object {
                "borderColor": "#bdbfff",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#bdbfff",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Create
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 50,
                "borderWidth": 1,
                "height": 50,
                "justifyContent": "center",
                "marginTop": 25,
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "#EEEEEE",
                "borderColor": "#EEEEEE",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#666",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Cancel
          </Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "marginBottom": 200,
              "marginTop": 25,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#5D61FF",
                "fontSize": 15,
              }
            }
          >
            Back to previous step
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</ScrollView>
`;
